numpy/core/tests/test_scalar_methods.py

```diff
@@ -0,0 +1,72 @@
+"""
+Test the scalar constructors, which also do type-coercion
+"""
+from __future__ import division, absolute_import, print_function
+
+import fractions
+import numpy as np
+
+from numpy.testing import (
+    run_module_suite,
+    assert_equal, assert_almost_equal, assert_raises, assert_warns,
+    dec
+)
+
+float_types = [np.half, np.single, np.double, np.longdouble]
+
+def test_floatasratio():
+    rstate = np.random.RandomState()
+
+    # derived from the cpython test
+    for ftype in float_types:
+        fi = np.finfo(ftype)
+
+        for f, ratio in [
+                (0.875, (7, 8)),
+                (-0.875, (-7, 8)),
+                (0.0, (0, 1)),
+                (11.5, (23, 2)),
+            ]:
+            assert_equal(ftype(f).as_integer_ratio(), ratio)
+
+        for i in range(10000):
```

Comment:
I realize the tests are from cpython, but I still dislike throwing random numbers at tests - it just means that things become non-reproducible. Why not explicitly pick the smallest and largest representable number? And do `2.1` for all the types.

Reply:
Do you want me to fix the random seed?

Note the result when applied to 2.1 is different for each type (and platform dependent for float128)

Reply:
Do you want me to fix the random seed?

Note the result when applied to 2.1 is different for each type (and platform dependent for float128)

Reply:
I'd just prefer not to use random numbers at all, and not 1000 of them either (needless increase of test time), i.e., just loop over some 10 examples chosen with some care (just testing as you do now that the result is consistent).

And, yes, you're right, for 2.1 the results would be different -- it's fine not to include 128 for that, given the platform dependence.

Reply:
I think my ability to choose relevant corner cases is very limited, so I'd prefer to keep the randomness. I've reduced the number of iterations from 10000 to 1000, seeded the randomstate, and added explicit tests for 2.1.

Note that the largest representable number is not testable, as its denominator may be larger than itself. The relevant tests are the ones that use lots of mantissa bits.

Reply:
OK, this was not meant to be a big deal.